api/baseline-api-v1.0.0.yaml

```diff
@@ -0,0 +1,4382 @@
+      type: object
+      properties:
+        witness:
+          $ref: '#/components/schemas/Witness'
+        proof:
+          type: string
+          description: The hex-formatted proof to verify
+        store:
+          type: boolean
+          description: when true, the proof hash is written to the configured circuit store
+      example:
+        witness:
+          X: '3'
+          'Y': '35'
+        proof: 88b8d340795f4258c1f2c0ed8478b44859c90472af9ccdc5c54b02c85bcd879d9ddf3a3fb7c6926ea247c0b1c411abe2f5ba9c8deba60ab476428272767a761d1e14e343208024326429e361ebbc4c11daf5f16dbb7f7f4679cf85b17b59b1bdaa379a82ddab4dcbb7cd2bc32cdebadc6825c72d5f8e624f6d0f6b07906a8248
+    Wallet:
+      title: Wallet
+      type: object
+      properties:
+        id:
+          type: string
+          description: UUID for a unique `Wallet`. _UUID as specified by RFC4122_
+          format: uuid
+          readOnly: true
+        created_at:
+          type: string
+          description: Date and time of `Wallet` creation
+          readOnly: true
+        network_id:
+          type: string
+          format: uuid
+          readOnly: false
+          description: UUID for the target `Network`. _UUID as specified by RFC4122_
+        user_id:
+          type: string
+          description: UUID for the `User` that created a `Wallet`. _UUID as specified by RFC4122_
+          readOnly: true
+        vault_id:
+          type: string
+          description: UUID for the `Vault` that created a `Wallet`. _UUID as specified by RFC4122_
+          format: uuid
+          readOnly: true
+        key_id:
+          type: string
+          description: UUID of private key used to create `Wallet`
+          format: uuid
+          readOnly: true
+        purpose:
+          type: integer
+          description: Purpose is a constant set to 44' (or 0x8000002C) following the BIP43 recommendation. It indicates that the subtree of this node is used according to this specification. Hardened derivation is used at this level
+          format: string
+          readOnly: false
+        public_key:
+          type: string
+          description: Public signing key of `Wallet`
+          readOnly: true
+      description: hierarchical deterministic (HD) wallet in accordance with BIP44
+      example:
+        id: bbc1195e-d131-4b2f-b1a8-1b6f34b55d3e
+        created_at: 2021-03-30T02:50:17.155Z
+        user_id: 7c8fe6f1-38c3-4da1-b4b7-7591c6d0ca7c
+        vault_id: 190822c9-62f5-4caf-a419-df735a793b2f
+        key_id: db9e6e21-23b0-497d-a02c-8a8813f8bf2d
+        purpose: 44
+        public_key: xpub661MyMwAqRbcGdYXwwnwcnrH51AxyF3kkTeEt6iZfNDXsg4MLPgV7bEZ2v4uCx9djAboy6vSv3VbHPc6hf4Do8wb7FpJCuG7aMBwH2QgLJu
+    Witness:
+      title: Witness
+      required:
+        - X
+        - 'Y'
+      type: object
+      properties:
+        X:
+          type: string
+        'Y':
+          type: string
+      description: Public and private inputs used to generate the proof
```

Comment:
you have multiple inputs into a circuit, so array?

Reply:
Actually this is an object where named params are mapped reflectively to the circuit params.